src/custom-pv-components/pdb-navigation.js

```diff
@@ -17,25 +17,58 @@ const height = 40,
   };
 
 class ProtvistaPdbNavigation extends ProtvistaNavigation {
-  
+
   connectedCallback() {
     this.style.display = 'block';
     this.style.width = '100%';
     this.width = this.offsetWidth;
 
     this._offset = parseFloat(this.getAttribute('offset')) || 0;
+    this._length = parseInt(this.getAttribute('length'));
+    this._displaystart = parseInt(this.getAttribute('displaystart')) || (this._offset > 0) ? this._offset : 1;
+    this._displayend = parseInt(this.getAttribute('displayend')) || (this._offset > 0) ? (this._length + this._offset - 1) : this._length;
+    this._highlightstart = parseInt(this.getAttribute('highlightstart'));
+    this._highlightend = parseInt(this.getAttribute('highlightend'));
+    this._highlightintervals = this.getAttribute("highlightintervals");
 
-    this._length = parseFloat(this.getAttribute('length'));
-    this._displaystart = parseFloat(this.getAttribute('displaystart')) || (this._offset > 0) ? this._offset : 1;
-    this._displayend = parseFloat(this.getAttribute('displayend')) || (this._offset > 0) ? (this._length + this._offset - 1) : this._length;
-    this._highlightStart = parseFloat(this.getAttribute('highlightStart'));
-    this._highlightEnd = parseFloat(this.getAttribute('highlightEnd'));
+    this._highlightedFragments = [];
 
     this._onResize = this._onResize.bind(this);
+    this._updateNavRuler = this._updateNavRuler.bind(this);
 
     this._createNavRuler();
   }
 
+  static get observedAttributes() {
+    return ['length', 'displaystart', 'displayend', 'highlightstart', 'highlightend', 'width', 'highlightintervals'];
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+    if (oldValue !== newValue) {
+      if (name === "highlightintervals") this[`_${name}`] = newValue;
+      else if (name === "offset") this[`_${name}`] = parseFloat(newValue);
+      else this[`_${name}`] = parseInt(newValue);
+
+      this._updateNavRuler();
+    }
+  }
+
+  _onResize() {
+    this.width = this.offsetWidth;
+    if (this.width - padding.right < 0) return; //rect cannot have negative width
+    this._x = this._x.range([padding.left, this.width - padding.right]);
+
+    this._svg.attr('width', this.width);
+
+    this._axis.call(this._xAxis);
+
+    this._viewport.extent([[padding.left, 0], [this.width - padding.right, height * 0.51]]);
+
+    this._brushG.call(this._viewport);
+
+    this._updateNavRuler();
+  }
+
```

Comment:
[subjective, nit-picking] Typically, we use blank lines to separate logic blocks. Here I am not sure what they separate, I'd use less of them.

Reply:
There are a bunch of methods that I have added that are from the super class just to override some statements buggy or to add more functionality (in this case d3 error with negative value). So I also kept this blank lines, but I think the same, idk why they put here blank lines after almost each statement. Should I remove them, or should I keep the legacy code?

Reply:
Let's keep them as legacy code.